central/reports/scheduler/v2/scheduler_impl.go

```diff
@@ -322,14 +325,30 @@ func (s *scheduler) queuePendingReports() {
 	}
 
 	for _, snap := range pendingReports {
-		reportReq, err := validation.ValidateAndGenerateReportRequest(s.reportConfigDatastore, s.collectionDatastore, s.notifierDatastore,
-			snap.GetReportConfigurationId(), snap.GetRequester(), snap.GetReportStatus().GetReportNotificationMethod(),
-			snap.GetReportStatus().GetReportRequestType())
+		_, found, err := s.reportConfigDatastore.GetReportConfiguration(scheduledCtx, snap.GetReportConfigurationId())
 		if err != nil {
 			log.Errorf("Error rescheduling pending report job for report config ID '%s': %s", snap.GetReportConfigurationId(), err)
 			continue
 		}
-		_, err = s.SubmitReportRequest(scheduledCtx, reportReq, true)
+		if !found {
+			log.Warnf("Report configuration with ID %s had pending reports but the configuration no longer exists",
```

Comment:
1. This should be error.
2. The report should be removed from the queue if configuration cannot be found.  Does `s.removeFromRunningReportConfigs(req.ReportSnapshot.GetReportConfigurationId())` handle it>

Reply:
Actually this should never happen because we have a foreign key on `report_configuration_id` in report_snapshot.proto .

`string   report_configuration_id  = 2 [(gogoproto.moretags) = 'search:"Report Configuration ID" sql:"fk(ReportConfiguration:id)"'];`

Since all pending reports are fetched from `report_snapshots` table, they are not in queue at this point. If reportConfig is deleted, then the reportSnapshot should be deleted from the database too.

Reply:
If it is unexpected, then change the logging to `utils.Should`

Reply:
Thinking more about it, it may happen in a rare race condition when a config gets deleted while we are looping through the snapshots here. But when that happens, the job would have been removed from the db and it won't be in queue. The only place that job would exist is in this loop. So, I think we are safe with just logging an error and skipping the job.